src/Translator.php

```diff
@@ -0,0 +1,512 @@
+                if ($dirty !== $cleaned) {
+                    $num++;
+                    file_put_contents($sourceFile, $cleaned);
+                }
+            }
+            $this->log("<info>{$num}</info> yml files cleaned");
+        }
+    }
+
+    /**
+     * Merge any missing keys from old json content into json files
+     */
+    private function mergeJson(): void
+    {
+        $this->log('Merging local json files');
+        foreach ($this->originalJson as $path => $contentJson) {
+            // If there are any keys in the original json that are missing now, add them back in.
+            $parsedJson = $this->jsonDecode(file_get_contents($path));
+            $contentJson = array_merge($contentJson, $parsedJson);
+            file_put_contents($path, $this->jsonEncode($contentJson));
+        }
+        $this->log('Finished merging ' . count($this->originalJson) . ' json files');
+    }
+
+    /**
+     * Run text collector on the given modules
+     */
+    private function collectStrings(): void
+    {
+        $this->log("Running i18nTextCollectorTask");
+        $modulesNames = [];
+        foreach ($this->modulePaths as $modulePath) {
+            $modulesNames[] = $this->getModuleName($modulePath);
+        }
+        $module = urlencode(implode(',', $modulesNames));
+        $site = rtrim($this->txSite, '/');
+        $this->exec("wget $site/dev/tasks/i18nTextCollectorTask?flush=all&merge=1&module=$module");
+    }
+
+    /**
+     * Push source updates to transifex
+     */
+    private function transifexPushSource(): void
+    {
+        $this->log('Pushing updated sources to transifex');
+        if ($this->isDevMode) {
+            $this->log('Not pushing to transifex because TX_DEV_MODE is enabled');
+            return;
+        }
+        foreach ($this->modulePaths as $modulePath) {
+            $this->exec('tx push -s', $modulePath);
+        }
+    }
+
+    /**
+     * Commit changes for all modules
+     */
+    private function gitCommitPushAndPullRequest(): void
+    {
+        $this->log('Committing translations to git');
+        foreach ($this->modulePaths as $modulePath) {
+            // Get endpoint
+            $remote = $this->exec('git config --get remote.origin.url', $modulePath);
+            $remote = str_replace('git@github.com:', 'https://github.com/', $remote);
+            if (!preg_match('#^https://github.com/([^/]+)/(.+?)$#', $remote, $matches)) {
+                throw new Exception("Invalid git remote $remote");
+            }
+            array_shift($matches);
+            list($account, $repo) = $matches;
+            $repo = preg_replace('#\.git$#', '', $repo);
+            $endpoint = "https://api.github.com/repos/$account/$repo/pulls";
+
+            // Add remote
+            if (!in_array('tx-ccs', explode("\n", $this->exec('git remote', $modulePath)))) {
+                $this->exec("git remote add tx-ccs git@github.com:creative-commoners/$repo.git", $modulePath);
```

Comment:
This is intentionally hardcoded to creative-commoners. It's only intended to be used by Silverstripe CMS Squad, similar to how cow is now